org.lflang/src/org/lflang/generator/ReactionInstance.java

```diff
@@ -499,6 +500,22 @@ public TimeValue assignLogicalExecutionTime() {
         return this.let = let;
     }
 
+    /**
+     * Find the inherited deadline which is the least deadline of all
+     * downstream reactions. This inherited deadline is used to assign
+     * priority to the reaction
+     */
+    // FIXME: Needs to be recursive to support chains.
+    public TimeValue getInheritedDeadline() {
```

Comment:
Shouldn't this also check against this.deadline?

Reply:
It is checked somewhere else. But I will revise the implementation and use:
ReactionInstance.declaredDeadling: Deadline specified by user.
ReactionInstance.deadine for the "inferred" deadline which is either MIN(downstream deadlines) or declaredDeadline.